server/rooms.ts

```diff
@@ -1130,11 +1126,13 @@ export class GlobalRoomState {
 				if (!p1userid || !p2userid) continue;
 				if (!p1userid.startsWith(usernameFilter) && !p2userid.startsWith(usernameFilter)) continue;
 			}
-			if (skipCount && skipCount--) continue;
-
 			rooms.push(room);
 		}
 
+		if (rooms.length > 150 && !formatFilter && !eloFilter && !usernameFilter) {
+			rooms = rooms.slice(-150);
+		}
+
 		const roomTable: {[roomid: string]: BattleRoomTable} = {};
 		for (let i = rooms.length - 1; i >= rooms.length - 100 && i >= 0; i--) {
```

Comment:
Specifically, what the code was doing was searching the most recent 150 battles, under the assumption that it will contain at least 100 public battles. The actual code to grab 100 battles is here.

Reply:
Yeah, I understood there would be performance implications but didnt notice there was a second level of filtering below . Do we want to just increase the 150 to like 1500 or something more conservative (and leave a comment explaining where the number came from...)? On main I was noticing like ~6000 rooms, what motivated this additional skip optimization?

Reply:
I used to do a lot of premature optimization. It's probably unnecessary.